usaspending_api/spending_explorer/v2/filters/type_filter.py

```diff
@@ -106,43 +100,50 @@ def type_filter(_type, filters, limit=None):
-        elif key == "quarter":
-            if value in ("1", "2", "3", "4"):
-                fiscal_quarter = int(value)
-            else:
-                raise InvalidParameterException(
-                    "Incorrect value provided for quarter parameter. Must be a string between 1 and 4"
-                )
-
-    if fiscal_year:
-        fiscal_date, fiscal_quarter = generate_last_completed_fiscal_quarter(
-            fiscal_year=fiscal_year, fiscal_quarter=fiscal_quarter
-        )
+    if "fy" not in filters:
+        raise InvalidParameterException('Missing required parameter "fy".')
 
-    # Recipient, Award Queryset
+    if "quarter" not in filters:
+        raise InvalidParameterException('Missing required parameter "quarter".')
+
+    try:
+        fiscal_year = int(filters["fy"])
+        if fiscal_year < 1000 or fiscal_year > 9999:
+            raise InvalidParameterException('Incorrect Fiscal Year Parameter, "fy": "YYYY"')
+    except ValueError:
+        raise InvalidParameterException('Incorrect or Missing Fiscal Year Parameter, "fy": "YYYY"')
+
+    if filters["quarter"] not in ("1", "2", "3", "4", 1, 2, 3, 4):
+        raise InvalidParameterException("Incorrect value provided for quarter parameter. Must be between 1 and 4")
+    fiscal_quarter = int(filters["quarter"])
+
+    submission_window = DABSSubmissionWindowSchedule.objects.filter(
+        submission_fiscal_year=fiscal_year,
+        submission_fiscal_quarter=fiscal_quarter,
+        is_quarter=True,
+        submission_reveal_date__lte=datetime.now(timezone.utc),
+    ).first()
+    if submission_window is None:
+        raise InvalidParameterException(f"No submission data available for FY{fiscal_year}Q{fiscal_quarter}.")
+
+    fiscal_date = submission_window.period_end_date
+    fiscal_period = submission_window.submission_fiscal_month
+
+    # transaction_obligated_amount is summed across all periods in the year up to and including the requested quarter.
     alt_set = (
-        FinancialAccountsByAwards.objects.all()
-        .exclude(transaction_obligated_amount__isnull=True)
+        FinancialAccountsByAwards.objects.exclude(transaction_obligated_amount__isnull=True)
         .exclude(transaction_obligated_amount="NaN")
-        .filter(submission__reporting_fiscal_quarter=fiscal_quarter)
+        .filter(submission__reporting_fiscal_quarter__lte=fiscal_quarter)
         .filter(submission__reporting_fiscal_year=fiscal_year)
         .annotate(amount=Sum("transaction_obligated_amount"))
     )
 
-    # Base Queryset
+    # obligations_incurred_by_program_object_class_cpe is picked from the final period of the quarter.
     queryset = (
-        FinancialAccountsByProgramActivityObjectClass.objects.all()
-        .exclude(obligations_incurred_by_program_object_class_cpe__isnull=True)
-        .filter(submission__reporting_fiscal_quarter=fiscal_quarter)
+        FinancialAccountsByProgramActivityObjectClass.objects.exclude(
```

Comment:
Could switch the exclude logic and add to the filter
`obligations_incurred_by_program_object_class_cpe__isnull=False`